crates/l2/sequencer/mod.rs

```diff
@@ -161,12 +163,24 @@ pub async fn start_l2(
     }
 
     if cfg.monitor.enabled {
-        task_set.spawn(monitor::start_monitor(
+        let cancelation_token = CancellationToken::new();
+        EthrexMonitor::spawn(
             shared_state.clone(),
             store.clone(),
             rollup_store.clone(),
-            cfg.clone(),
-        ));
+            &cfg,
+            cancelation_token.clone(),
+        )
+        .await?;
+
+        task_set.spawn(async move {
+            tokio::select! {
+                _ = cancelation_token.cancelled() => {
+                    info!("Monitor has been cancelled");
+                    Ok(())
+                }
+            }
+        });
```

Comment:
A `tokio::select!` with only one branch can be replaced with the branch's future await, right?

Isn't this equivalent?

```suggestion
        task_set.spawn(async move {
            cancelation_token.cancelled().await;
            info!("Monitor has been cancelled");
            Ok(())
        });
```

In any case, I think spawning an awaiting task just to log when the Monitor has been cancelled seems unnecessary.

Maybe we can log the cancellation within the Monitor itself, that way we also won't need the cancellation token at all

Reply:
The task is not just to log the cancellation, the idea is that when this task finishes, the whole process also does.
This is done by the later `join_next` on that file, which makes this function end.
I'll do the change to make it not use the select

Reply:
Done